staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/subresource.go

```diff
@@ -0,0 +1,180 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fieldmanager
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/internal"
+	"sigs.k8s.io/structured-merge-diff/v4/fieldpath"
+)
+
+// SubresourceScaler manages the conversion of managed fields between a main
+// resource and subresources
+type SubresourceScaler struct {
+	parentEntries           []metav1.ManagedFieldsEntry
+	parentGroupVersion      schema.GroupVersion
+	subresourceGroupVersion schema.GroupVersion
+
+	parentPath      fieldpath.Path
+	subresourcePath fieldpath.Path
+}
+
+// ScalePathMappings maps a group/version to the replicas path in a scale subresource
+type ScalePathMappings map[schema.GroupVersion][]interface{}
+
+// NewSubresourceScaler creates a new SubresourceScaler
+func NewSubresourceScaler(parentEntries []metav1.ManagedFieldsEntry, parentGroupVersion schema.GroupVersion, parentPathPieces []interface{}) (*SubresourceScaler, error) {
+	h := SubresourceScaler{
+		parentEntries:      parentEntries,
+		parentGroupVersion: parentGroupVersion,
+	}
+
+	// The assumption is that scale objects have autoscaling/v1 group version and
+	// corresponding .spec.replicas path
+	h.subresourceGroupVersion = schema.GroupVersion{Group: "autoscaling", Version: "v1"}
+	subresourcePath, err := fieldpath.MakePath("spec", "replicas")
+	if err != nil {
+		return nil, err
+	}
+	h.subresourcePath = subresourcePath
+
+	parentPath, err := fieldpath.MakePath(parentPathPieces...)
```

Comment:
Compute the paths outside this function, and you probably need to receive a mapping. Maybe create the mapping directly to path so that they don't have to be recomputed (and checked for errors) everytime.

Since all this function does is compute these various things, I don't think it's super useful, people should be able to construct the object directly.